lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -568,7 +568,7 @@ def authenticate_via_krb5_ccache_credential_tgt(credential, options = {})
       value: credential.keyblock.data.value
     )
 
-    etypes = Set.new([ticket.enc_part.etype])
+    etypes = Set.new([credential.keyblock.enctype.value])
```

Comment:
There's other lines like `etypes = Set.new([ticket.enc_part.etype])` in this file that I believe are wrong too. Will try to replicate those edgecases and update once confirmed.